lib/rb_shift/client.rb

```diff
@@ -62,11 +64,16 @@ def projects(update = false)
     end
 
     def execute(command, **opts)
-      `oc --server="#{@url}" --token="#{@token}" #{command}  #{unfold_opts opts}`
+      command = "oc --server=\"#{@url}\" --token=\"#{@token}\" #{command}  #{unfold_opts opts}"
+      Open3.popen3 command do |_, _, stderr, wait_thrd|
+        err_code = wait_thrd.value
+        raise InvalidCommandError unless stderr.gets.nil? && err_code.success?
+      end
     end
 
     def wait_project_deletion(project_name, timeout = 1)
-      sleep timeout until projects(true).find { |v| v.name == project_name }.nil?
+      sleep timeout until get('projects').find { |v| v[:metadata][:name] == project_name }.nil?
```

Comment:
Yop.... It doesn't really matter which one. I just think something like 

``` sleep timout while get('project').detect {}``` would be more readable -- also it's kinda common to rely on null/non-null equivalence in dynamic languages like ruby/groovy/python (empty list essentially is equal to either nil or false).